### PR TITLE
LITE-26312: Plugin flake8-pyproject added, updated pyproject.toml and…

### PR DESCRIPTION
**Pull request by gab832**
… github actions

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,34 +28,34 @@ keywords = [
...
+flake8-pyproject = "^1.2.2"
 
 [build-system]
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.pytest.ini_options]
-testpaths = "tests"
+testpaths = ["tests"]
 log_cli = true
```
Comment (ffaraone) [inline pyproject.toml]: By default log_cli should be false

File: pyproject.toml [hunk 2]
```diff
@@ -15,7 +15,6 @@ classifiers = [
     "Intended Audience :: Developers",
     "License :: OSI Approved :: Apache Software License",
     "Operating System :: Unix",
-    "Programming Language :: Python :: 3.6",
     "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
```
Comment (ffaraone) [inline pyproject.toml:20]: Probably we should add here trove classifier for 3.10 and 3.11

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -14,14 +14,14 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        python-version: ['3.7', '3.8', '3.9', '3.10']
```
Comment (ffaraone) [inline .github/workflows/build.yml]: We should add here 3.11

File: pyproject.toml [hunk 4]
```diff
@@ -29,34 +28,34 @@ keywords = [
...
-flake8 = "~3.8"
-flake8-bugbear = "~20"
+coverage = {extras = ["toml"], version = "^5.3,<7"}
+flake8 = ">=5,<=6"
+flake8-bugbear = ">=20,<23"
 flake8-cognitive-complexity = "^0.1"
-flake8-commas = "~2.0"
+flake8-commas = "~2.1"
 flake8-future-import = "~0.4"
 flake8-import-order = "~0.18"
```
Comment (ffaraone) [inline pyproject.toml]: We should move from flake8-import-order to flake8-isort

**Review by ffaraone — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 5]
```diff
@@ -29,34 +28,34 @@ keywords = [
 
 
 [tool.poetry.dependencies]
-python = "^3.6"
+python = ">=3.7,<4"
 lark-parser = "0.11.0"
 cachetools = ">=4.2.4"
 python-dateutil = ">=2.8.2"
 
 [tool.poetry.dev-dependencies]
```
Comment (ffaraone) [inline pyproject.toml:38]: `tool.poetry.group.test.dependencies`

**Review by ffaraone — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 5 above)
Comment (gab832) [reply pyproject.toml:38]: Not allowed because of the poetry version allowed for the python 3.7

Comment (ffaraone) [reply pyproject.toml:38]: 👍

File: pyproject.toml [hunk 6]
```diff
@@ -80,3 +80,23 @@ exclude_lines = [
     "raise NotImplementedError",
     "if __name__ == .__main__.:",
 ]
+
+[tool.flake8]
+exclude = [".idea", ".git", "venv*/", ".eggs/", "*.egg-info"]
+max-line-length = 100
+show-source = true
+application-import-names = ["py_rql", "tests"]
```
Comment (ffaraone) [inline pyproject.toml]: This is for flake8-import-order that is no longer used, please remove it

**Review by ffaraone — CHANGES_REQUESTED** [review]

**Review by ffaraone — APPROVED** [review]

**Pull request merged**